core/amber/src/main/scala/edu/uci/ics/texera/web/resource/WorkflowWebsocketResource.scala

```diff
@@ -172,25 +193,30 @@ class WorkflowWebsocketResource {
       context
     )
 
-//    texeraWorkflowCompiler.init()
+    //    texeraWorkflowCompiler.init()
     val violations = texeraWorkflowCompiler.validate
     if (violations.nonEmpty) {
       send(session, WorkflowErrorEvent(violations))
       return
     }
 
+    clearCache(session)
+
     val workflow = texeraWorkflowCompiler.amberWorkflow
     val workflowTag = WorkflowIdentity(jobID)
 
     val eventListener = ControllerEventListener(
       workflowCompletedListener = completed => {
-        sessionResults.remove(session.getId)
         sessionResults.update(session.getId, completed.result)
```

Comment:
you removed a line

Reply:
I moved it into the `clearCache` function.